biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -585,9 +582,9 @@ subroutine zero_cohort(cc_p)
 
     currentcohort%year_net_uptake(:) = 999._r8 ! this needs to be 999, or trimming of new cohorts will break. 
     currentcohort%ts_net_uptake(:)   = 0._r8
-    currentcohort%seed_prod          = 0._r8
     currentcohort%fraction_crown_burned = 0._r8 
```

Comment:
I'm glad you brought this up @jkshuman , this has been on my mind too.
In my opinion, every variable, upon allocation should be "nan'd".  
 The power of Nan'ing things, is that it makes it clear the variable has not been set since it was defined, which is so useful for debugging and checking the code.  I don't like the idea of doing a blanket "zero'ing" of all variables after allocation.  Once we zero things, its tough to tell at any given time in the call sequence, if a variable is zero because an actual process wants it to be zero, or because we never called a process to modify it.
But there are some things, like variables that are accumulated over several different processes (like mass conservation checks, or fluxes, diagnostics), that I do like to see zero'd as a group.  But, outside of this type of scenario, in my opinion, I think we should avoid blanket zero'ing (particularly prognostic (time-integrated, based on fluxes) variables, unless there is intent, like zero'ing iniital litter mass in a spin-up run).


Performing an audit of all cohort and patch level variables, and addressing their nan'ing and zero'ing has been on my to-do list for a while.